File: Demos/Datasnap/Server/proxy/java_android/com/embarcadero/javaandroid/TDBXReaderValue.java
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

package com.embarcadero.javaandroid;

/**
 * Wraps the {@link TDBXReader} and allows it to be null.
 */

public class TDBXReaderValue extends DBXValue {
	protected boolean ValueNull = false;
	
	public TDBXReaderValue() {
		super();
		setDBXType(DBXDataTypes.TableType);
	}
	
	public void setNull() {
		ValueNull = true;
		objectValue = null;
	}

	public boolean isNull() {
		return ValueNull;
	}
	
	@Override
	public void SetAsTable(TableType Value) {
		objectValue = (Object) Value;
	}

	@Override
	public Object GetAsTable() throws DBXException {
		return objectValue;
	}
	
	public TDBXReader GetAsDBXReader() throws DBXException {
		return (TDBXReader) objectValue;
	}
	
	public void SetAsDBXReader(TableType Value) {
		SetAsTable(Value);
	}

}

File: Demos/Datasnap/Server/proxy/objectivec_ios81/Proxy/TDBXReaderValue.m
//*******************************************************
//
//               Delphi DataSnap Framework
//
// Copyright(c) 1995-2011 Embarcadero Technologies, Inc.
//
//*******************************************************

#import "TDBXReaderValue.h"



@implementation TDBXReaderValue
-(id) init {
	self = [super init];
	if (self) {
		[self setDBXType:TableType];
		ValueNull =  NO;
	}
	return self;
}
-(void) SetNull {
	ValueNull = YES;
	objectValue = nil;
}

-(bool) isNull {
	return ValueNull;
}


-(void) SetAsTable:(id<TableType>)value{
	objectValue = value;
	[self setDBXType:TableType];}


-(id<TableType>) GetAsTable{
	[self checkCurrentDBXType:TableType];
	return objectValue ;
}

-(TDBXReader *) GetAsDBXReader {
	return (TDBXReader *) objectValue;
}

-(void) SetAsDBXReader:(id<TableType>) value {
	[self SetAsTable:value];
}

@end
